### PR TITLE
[DEC-2081] Audit app/middleware/middleware.go

### PR DESCRIPTION
**Pull request by lcwik**
Changes:
* swap to use test cases
* ensure that the middle ware logger is restored after the test finishes

### CODE REVIEW

File: protocol/app/middleware/middleware_test.go [hunk 1]
```diff
@@ -1,68 +1,73 @@
...
+			}()
 
-	func() {
-		defer func() {
-			if r := recover(); r != nil {
-				handler := middleware.NewRunTxPanicLoggingMiddleware()
-				err := handler(r)
-				require.Nil(t, err)
+			if tc.expectedLogs == nil {
+				require.Equal(t, "", buf.String())
```
Comment (ttl33) [inline protocol/app/middleware/middleware_test.go:66]: nit: does `require.Empty()` work here?

File: protocol/app/middleware/middleware_test.go [hunk 2]
```diff
@@ -1,68 +1,73 @@
...
-		"Error",
-		"runTx panic'ed with a problem",
-		"stack trace",
-		mock.Anything,
-	).Return(nil).Once()
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			// Restore the old logger after the test runs
+			oldLogger := middleware.Logger
+			defer func() { middleware.Logger = oldLogger }()
```
Comment (ttl33) [inline protocol/app/middleware/middleware_test.go:49]: for my understanding, why do we need to restore the logger? is it to reset the logger for each test run?

**Review by ttl33 — APPROVED** [review]

File: protocol/app/middleware/middleware_test.go (hunk 2 above)
Comment (lcwik) [reply protocol/app/middleware/middleware_test.go:49]: middleware.Logger is a global variable, clarified in comment

File: protocol/app/middleware/middleware_test.go (hunk 1 above)
Comment (lcwik) [reply protocol/app/middleware/middleware_test.go:66]: It does, updated.

**Pull request merged**